crates/wasmtime/src/config.rs

```diff
@@ -103,14 +106,56 @@ pub struct Config {
     pub(crate) force_memory_init_memfd: bool,
 }
 
+/// User-provided configuration for the compiler.
+#[cfg(compiler)]
+#[derive(Debug, Clone)]
+pub(crate) struct CompilerConfig {
```

Comment:
I think it's ok to remove `pub(crate)` on all these methods and related fields since it should be able to isolate all the usages to just this file.